src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -542,33 +542,17 @@ void ImageReplayer<I>::on_start_fail_finish(int r)
   Context *on_stop_finish(nullptr);
   {
     Mutex::Locker locker(m_lock);
-    if (m_stop_requested) {
-      assert(r == -EINTR);
-      dout(20) << "start interrupted" << dendl;
-      m_state = STATE_STOPPED;
-      m_stop_requested = false;
+    if (r < 0) {
+      derr << "start failed: " << cpp_strerror(r) << dendl;
     } else {
-      assert(m_state == STATE_STARTING);
-      dout(20) << "start failed" << dendl;
-      m_state = (r < 0) ? STATE_UNINITIALIZED : STATE_STOPPED;
+      dout(20) << "start interrupted" << dendl;
```

Comment:
I mean, that previously, when I did:
```
start
stop (before start is completed)
```
I got "start interrupted" message in the log.

Now I see "start interrupted" message in the secondary mirror log periodically on every its try to bootstrap (for "remote image is non-primary or local image is primary" case).

I just say that the behavior is changed, personally I liked more as it was before, but I have no strong opinion here.